internal/dag/gatewayapi_processor.go

```diff
@@ -59,36 +64,115 @@ type matchConditions struct {
 func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 	p.dag = dag
 	p.source = source
+	p.matchingHTTPRoutes = make(map[int][]*gatewayapi_v1alpha1.HTTPRoute)
+	p.matchingTLSRoutes = make(map[int][]*gatewayapi_v1alpha1.TLSRoute)
+	p.listenerSecret = make(map[int]*Secret)
 
 	// reset the processor when we're done
 	defer func() {
 		p.dag = nil
 		p.source = nil
 	}()
 
-	// Gateway must be defined for resources to be processed.
 	if p.source.gateway == nil {
-		p.Error("Gateway is not defined!")
```

Comment:
Ack. I changed error -> info.

Reply:
Sweet. Should we update the message while we're here? Something like the gateway is not found in the cluster since it would need to be defined already to get to this point.

Reply:
Maybe the error should say something like "No gateway pointing to the GatewayClass whatever could be found"?